sunpy/util/metadata.py

```diff
@@ -102,6 +101,17 @@ def __delitem__(self, key):
         OrderedDict.__delitem__(self, key.lower())
         self._prune_keycomments()
 
+    def item_hash(self):
+        """
+        Create a hash based on the stored items.
+
+        This relies on all the items themselves being hashable. For this reason
```

Comment:
Probably worth failing back to the old behaviour somehow if someone has put something unhashable in the meta dict manually.